From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ricardo Pardini <ricardo@pardini.net>
Date: Wed, 8 Jul 2026 23:21:21 +0200
Subject: mtd: spi-nor: Add support for Zbit ZB25LQ128

Add support for Zbit ZB25LQ128 (128M-bit) Serial Flash memory. Used on
the Seeed reComputer rk3576 DevKit board.

Datasheet: https://www.zbitsemi.com/upload/file/20220704/20220704135632_49171.pdf

Signed-off-by: Ricardo Pardini <ricardo@pardini.net>
---
 drivers/mtd/spi/spi-nor-ids.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mtd/spi/spi-nor-ids.c b/drivers/mtd/spi/spi-nor-ids.c
index 111111111111..222222222222 100644
--- a/drivers/mtd/spi/spi-nor-ids.c
+++ b/drivers/mtd/spi/spi-nor-ids.c
@@ -720,6 +720,8 @@ const struct flash_info spi_nor_ids[] = {
 	/* Zbit Semiconductor Inc. */
 	{ INFO("zb25vq128", 0x5e4018, 0, 64 * 1024, 256,
 	       SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+	{ INFO("zb25lq128", 0x5e5018, 0, 64 * 1024, 256,
+	       SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 #endif
 	{ },
 };
-- 
Armbian

